doc/source/user_guide/copy_on_write.rst

```diff
@@ -0,0 +1,213 @@
+.. _copy_on_write:
+
+{{ header }}
+
+*******************
+Copy-on-Write (CoW)
+*******************
+
+.. ipython:: python
+    :suppress:
+
+    pd.options.mode.copy_on_write = True
+
+Copy-on-Write was first introduced in version 1.5.0. Starting from version 2.0 most of the
+optimizations that become possible through CoW are implemented and supported. A complete list
+can be found at :ref:`Copy-on-Write optimizations <copy_on_write.optimizations>`.
+
+We expect that CoW will be enabled by default in version 3.0.
+
+CoW will lead to more predictable behavior since it is not possible to update more than
+one object with one statement, e.g. methods won't have side-effects. Additionally, through
+delaying copies as long as possible, the average performance will improve.
+
+Description
+-----------
+
+CoW means that any DataFrame or Series derived from another in any way always
+behaves as a copy. As a consequence, we can only change the values of an object
+through modifying the object itself. CoW disallows updating a DataFrame or a Series
+that shares data with another DataFrame or Series object inplace.
+
+This avoids side-effects when modifying values and hence, most methods can avoid
+actually copying the data and only trigger a copy when necessary.
+
+The following example will operate inplace with CoW:
```

Comment:
Not necessarily for this PR, but general comment: I think it might be easier to first show an example of the general idea of "not updating other dataframes" (i.e. the no-side-effects part), like you have a dataframe, select a subset, modify that subset -> with CoW that doesn't update the parent. 

I think that is something more common (and more essential to understand by out users), and easier to understand than the concept of whether the operation actually happened in place or not (depending on whether there are references). I would consider this as a more advanced part of the explanation, as you already need to understand the distinction between updating the object inplace (which still happens in either case) and updating the underlying values inplace. 

(And this distinction is currently also not really explained, eg the "will operate inplace with CoW" in the sentence above is actually ambiguous, as this operation is always "inplace" for the object)

Reply:
Yeah good point. Should do as a follow up?